java/src/main/java/io/opentracing/contrib/ServerTracingInterceptor.java

```diff
@@ -83,8 +82,15 @@ public ServerServiceDefinition intercept(BindableService bindableService) {
     ) {
         Map<String, String> headerMap = new HashMap<String, String>();
         for (String key : headers.keys()) {
-            String value = headers.get(Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER));
-            headerMap.put(key, value);
+            if (key.endsWith(Metadata.BINARY_HEADER_SUFFIX)) {
```

Comment:
If you think binary headers provide no value, then we can drop them. gRPC uses binary header to serialize some context. If it does add value for tracing, I am all for dropping them.